php_phongo.c

```diff
@@ -157,6 +154,7 @@ zend_class_entry* phongo_exception_from_mongoc_domain(uint32_t /* mongoc_error_d
 			return php_phongo_serverexception_ce;
 		case MONGOC_ERROR_CLIENT:
 		case MONGOC_ERROR_STREAM:
+			return php_phongo_connectionexception_ce;
```

Comment:
This may change behavior for the `MONGOC_ERROR_CLIENT` domain, since its case falls through.

Rather than move `case MONGOC_ERROR_CLIENT:` below this `return`, I'd propose you delete it and also remove all of the cases between this `return` and `default:`. This way, we only have cases for non-default return values.

Reply:
Additionally, can you remove the following lines down below?

```c
/* FIXME: We don't have the Exceptions mocked yet.. */
#if 0
	return phongo_ce_mongo_connection_exception;
#endif
```

`phongo_ce_mongo_connection_exception` does not exist and that's simply dead code.